crt_portal/cts_forms/forms.py

```diff
@@ -2028,63 +2028,104 @@ class PrintActions(Form):
 
-    def get_initial_values(self, record_query, keys):
-        """
-        Given a record query and a list of keys, determine if a key has a
-        singular value within that query. Used to set initial fields
-        for bulk update forms.
-        """
-        # make sure the queryset does not order by anything, otherwise
-        # we will have difficulty getting distinct results.
-        query = record_query.order_by()
-        for key in keys:
-            values = query.values_list(key, flat=True).distinct()
-            if values.count() == 1:
-                yield key, values[0]
+    def field_changed(self, field):
+        # if both are Falsy, nothing actually changed (None ~= "")
+        old = self.initial.get(field, None)
+        new = self.cleaned_data.get(field, None)
+        if not old and not new:
+            return False
+        return old != new
 
-    def __init__(self, query, *args, user=None, **kwargs):
+    @cached_property
+    def changed_data(self):
+        return [
+            field_name
+            for field_name
+            in super().changed_data
+            if self.field_changed(field_name)
+        ]
+
+    class Meta:
+        model = ReportDispositionBatch
+        fields = ['disposed_by', 'disposed_count', 'create_date', 'proposed_disposal_date']
+
+    def setup_disposed_by(self):
+        disposed_by = self.user
+        value = f'{disposed_by.first_name} {disposed_by.last_name}' if disposed_by.first_name and disposed_by.last_name else ''
+        self.fields['disposed_by'] = CharField(
+            label='Approving Official',
+            widget=CrtTextInput(
+                attrs={
+                    'class': 'usa-input',
+                    'name': 'disposed_by',
+                    'placeholder': 'Approving Official',
+                    'aria_label': 'Approving Official',
+                    'value': value,
+                    'label': 'Approving Official',
+                },
+            ),
+            required=True,
+            initial=disposed_by,
+        )
+
+    def setup_create_date(self):
+        create_date = datetime.today()
+        self.fields['create_date'] = CharField(
+            required=True,
+            label="Date",
+            initial=create_date,
+            widget=CrtTextInput(attrs={
+                'class': 'usa-input',
+                'name': 'create_date',
+                'placeholder': 'mm/dd/yyyy',
+                'aria_label': 'Date',
+                'value': create_date.strftime('%m/%d/%Y'),
+                'label': 'Date',
+            }),
+        )
+
+    def clean_disposed_by(self):
+        user = None
+        if 'disposed_by' not in self.cleaned_data:
+            return ''
+        name = self.cleaned_data['disposed_by'].split(' ')
+        if len(name) == 2:
+            user = User.objects.filter(first_name=name[0], last_name=name[1]).first()
+        if user:
+            return user
+        return ''
```

Comment:
this function looks like it returns a `Union[User, str]` - is that intentional?

Reply:
It should return either an empty string or a User object - could you explain what you mean by this?

Reply:
that is what I mean - as long as it's expected that we're mixing the User and string types, I'm good :D